api/src/events.py

```diff
@@ -0,0 +1,625 @@
+    return jsonify({"msg": "event updated"}), 200
+
+    
+@events_api.route('/addParticipant', methods=['POST'])
+@jwt_required
+def add_participant():
+
+    """Adds a participant to an event.
+
+    :reqjson event_name: event name
+    :reqjson user_id: the object ID of the user in the database
+
+    Example input:
+
+    .. sourcecode:: json
+
+    {
+        "event_name": "HopHacks_Fall_2022",
+        "user_id": "623e2e52eebe994953ba2f84"
+    }
+    
+    :status 200: Participant added
+    :status 400: Error with request or event doesn't exist
+
+    """
+
+    if not (all(field in request.json for field in ['event_name', 'user_id'])):
+        return Response('Invalid request', status=400)
+
+    event_name = request.json["event_name"]
+    user_id = request.json["user_id"]
+
+    event = {'event_name': event_name}
+    if len(list(db.events.find(event))) == 0:
+        return Response('Event Does Not Exist', status=400)
+
+    user = db.users.find_one({"_id": ObjectId(user_id)})
+    if user == None:
+        return Response('Event Does Not Exist', status=400) 
+    record = {
+        'user_id': user_id,
+        'username': user['username'],
+        'profile': user['profile'],
+        'status': 'applied'
+    }
+
+    db.events.update_one(event, {'$push': {'event_participants': record}, '$inc': {
+                         'num_registrations': 1}})
+
+    event_record = {
+        'event_name': event_name,
+        'status': 'applied'
+    }
+
+    db.users.update_one({"_id": ObjectId(user_id)}, {
+                        '$push': {'registrations': event_record}})
+
+    return jsonify({"msg": "participant added"}), 200
+
+
+    
+
+@events_api.route('/update_status', methods=['PUT'])
+@jwt_required
+def update_status():
+
+    """Updates the registration status of a user
+
+    :reqjson event_name: event name
+    :reqjson user_id: the object ID of the user in the database
+    :status: the new registration status of the user for this event
+
+    Example input:
+
+    .. sourcecode:: json
+
+    {
+        "event_name": "HopHacks_Fall_2022",
+        "user_id": "623e2e52eebe994953ba2f84",
+        "status": "rsvp"
+    }
+
+    :status 200: Registration status updated
+    :status 400: Invalid request or registration does not exist
+
+    """
+
+    if not (all(field in request.json for field in ['event_name', 'user_id', 'status'])):
+        return Response('Invalid request', status=400)
+
+    event_name = request.json['event_name']
+    user_id = request.json['user_id']
+    status = request.json['status']
+
+    if status not in ['applied', 'rsvp', 'canceled']:
+        return Response('Invalid request', status=400)
+
+
+    # Update event record
+    event_query = {'event_name': event_name}
+    event = db.events.find_one(event_query)
+    if event == None:
+        return Response('Event Does Not Exist', status=400)
+    registrations = event['event_participants']
+    found = False
+    for i in range(len(registrations)):
```

Comment:
maybe we can avoid fetch the entire list of registrations by referring to this https://www.mongodb.com/docs/manual/tutorial/query-arrays/#:~:text=To%20query%20if%20the%20array,value%3E%20is%20the%20element%20value.&text=To%20specify%20conditions%20on%20the,%3E%2C%20...%20%7D%20%7D